host/windows/syscall.c

```diff
@@ -324,6 +324,292 @@ __declspec(noreturn) static void _panic(
+        required_size = strlen(path) + current_dir_len + 1;
+    }
+
+    enclave_path = (char*)calloc(1, required_size);
+
+    const char* psrc = path;
+    const char* plimit = path + strlen(path);
+    char* pdst = enclave_path;
+
+    if (isalpha(*psrc) && psrc[1] == ':')
+    {
+        *pdst++ = '/';
+        *pdst++ = *psrc;
+        psrc += 2;
+    }
+    else if (*psrc == '/')
+    {
+        *pdst++ = '/';
+        *pdst++ = _getdrive() + 'a' - 1;
+    }
+    else if (*psrc == '.')
+    {
+        memcpy(pdst, current_dir, current_dir_len);
+        if (psrc[1] == '/' || psrc[1] == '\\')
+        {
+            pdst += current_dir_len;
+            psrc++;
+        }
+        else if (psrc[1] == '.' && (psrc[2] == '/' || psrc[2] == '\\'))
+        {
+            char* rstr = strrchr(
+                current_dir, '\\'); // getcwd always returns at least '\'
+            pdst += current_dir_len - (rstr - current_dir);
+            // When we shortend the curdir by 1 slash, we perform the ".."
+            // operation we could leave it in here, but at least sometimes this
+            // will allow a path that would otherwise be too long
+            psrc += 2;
+        }
+        else
+        {
+            // It is an incomplete which starts with a file which starts with .
+            // so we dont increment psrc at all
+            pdst += current_dir_len;
+            *pdst = '/';
+        }
+    }
+    else
+    {
+        // Still a relative path
+        memcpy(pdst, current_dir, current_dir_len);
+        pdst += current_dir_len;
+        *pdst++ = '/';
+    }
+
+    // Since we have to translater slashes, use a loop rather than memcpy
+    while (psrc < plimit)
+    {
+        if (*psrc == '\\')
+        {
+            *pdst = '/';
+        }
+        else
+        {
+            *pdst = *psrc;
+        }
+        psrc++;
+        pdst++;
+    }
+    *pdst = '\0';
+
+    if (current_dir)
+    {
+        free(current_dir);
+    }
+    return enclave_path;
+}
+
+// Allocates WCHAR* string which follows the expected rules for
+// enclaves comminication with the host file system API. Paths in the format
+// /<driveletter>/<item>/<item>  become <driveletter>:/<item>/<item>
+//
+// The resulting string, especially with a relative path, will probably contain
+// mixed slashes. We beleive Windows handles this.
+//
+// Adds the string "post" to the resulting string end
+//
+// The string  must be freed
+WCHAR* oe_syscall_path_to_win(const char* path, const char* post)
+{
+    size_t required_size = 0;
+    size_t current_dir_len = 0;
+    char* current_dir = NULL;
+    int pathlen = -1;
+    size_t postlen = 0;
+
+    if (!path)
+    {
+        return NULL;
+    }
+
+    if (strcmp(path, "/dev/null") == 0)
+    {
+        path = "NUL:";
+    }
+
+    pathlen = MultiByteToWideChar(CP_UTF8, 0, path, -1, NULL, 0);
+    if (post)
+    {
+        postlen = MultiByteToWideChar(CP_UTF8, 0, post, -1, NULL, 0);
```

Comment:
Check return value, if it did not succeed, it will be 0 and you can call GetLastError()